internal/app/machined/pkg/system/services/etcd.go

```diff
@@ -461,14 +460,9 @@ func (e *Etcd) argsForInit(ctx context.Context, r runtime.Runtime) error {
 		return err
 	}
 
-	p, err := platform.CurrentPlatform()
-	if err != nil {
-		return err
-	}
-
 	var upgraded bool
 
-	if p.Mode() != runtime.ModeContainer {
+	if r.State().Platform().Mode() != runtime.ModeContainer {
```

Comment:
it's a small unrelated change which got in, we don't need to build platform once again, as it's available via runtime state